adapters/sushiswap/src/sdk/lib.ts

```diff
@@ -0,0 +1,247 @@
+                    }
+                    token0Price
+                    token1Price
+                }
+            }
+        }`
+
+        const response = await fetch(V2_SUBGRAPH_URL, {
+            method: "POST",
+            body: JSON.stringify({ query }),
+            headers: { "Content-Type": "application/json" },
+        })
+        const { data: { liquidityPositions } } = await response.json();
+
+        result.push(...liquidityPositions.map((position: V2Position) => {
+            const { reserve0, reserve1 } = getV2PositionReserves(position)
+            return {
+                user: position.user.id,
+                token0: {
+                    address: position.pair.token0.id,
+                    balance: reserve0,
+                    symbol: position.pair.token0.symbol,
+                    usdPrice: +position.pair.token0Price
+                },
+                token1: {
+                    address: position.pair.token1.id,
+                    balance: reserve1,
+                    symbol: position.pair.token1.symbol,
+                    usdPrice: +position.pair.token1Price
+                }
+        }
+        }))
+
+        if (liquidityPositions.length < 1000) {
+            fetchNext = false;
+        } else {
+            skip += 1000;
+        }
+    }
+
+    return result
+}
+
+type V3Position = {
+    liquidity: string,
+    owner: string,
+    pool: {
+        sqrtPrice: string,
+        tick: string,
+        token0: {
+            id: string,
+            symbol: string,
+        },
+        token1: {
+            id: string,
+            symbol: string
+        }
+        token0Price: string,
+        token1Price: string
+    }
+    tickLower: {
+        tickIdx: string
+    },
+    tickUpper: {
+        tickIdx: string
+    },
+}
+
+const getV3PositionReserves = (position: V3Position) => {
+    const liquidity = +position.liquidity
+    const _sqrtPrice = +position.pool.sqrtPrice
+    const currentTick = +position.pool.tick
+    const tickLower = +position.tickLower.tickIdx
+    const tickUpper = +position.tickUpper.tickIdx
+
+    let reserve0 = 0n
+    let reserve1 = 0n
+
+    if (liquidity === 0) {
+        return {
+          reserve0,
+          reserve1,
+        }
+    }
+
+    const sqrtRatioA = Math.sqrt(1.0001 ** tickLower)
+    const sqrtRatioB = Math.sqrt(1.0001 ** tickUpper)
+    const sqrtPrice = _sqrtPrice / (2 ** 96)
+
+    if (currentTick < tickLower) {
+        reserve0 = BigInt(Math.floor(
+          liquidity * ((sqrtRatioB - sqrtRatioA) / (sqrtRatioA * sqrtRatioB)),
+        ))
+    } else if (currentTick >= tickUpper) {
+        reserve1 = BigInt(Math.floor(liquidity * (sqrtRatioB - sqrtRatioA)))
+    } else if (currentTick >= tickLower && currentTick < tickUpper) {
+        reserve0 = BigInt(Math.floor(liquidity * ((sqrtRatioB - sqrtPrice) / (sqrtPrice * sqrtRatioB))))
+        reserve1 = BigInt(Math.floor(liquidity * (sqrtPrice - sqrtRatioA)))
```

Comment:
@0xMasayoshi hey, we are looking to only calculate tvl that are in active range, can you edit this part to only get those are in range?

removing the first 2 if and keeping the last part should work. if (currentTick >= tickLower && currentTick < tickUpper)  ....

Reply:
hey @0xroll this has been fixed